pkg/apis/batch/validation/validation.go

```diff
@@ -490,11 +494,14 @@ func validateConcurrencyPolicy(concurrencyPolicy *batch.ConcurrencyPolicy, fldPa
 	return allErrs
 }
 
-func validateScheduleFormat(schedule string, timeZone *string, fldPath *field.Path) field.ErrorList {
+func validateScheduleFormat(schedule, oldSchedule string, timeZone *string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if _, err := cron.ParseStandard(schedule); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath, schedule, err.Error()))
 	}
+	if !strings.Contains(oldSchedule, "TZ") && strings.Contains(schedule, "TZ") {
```

Comment:
Does it mean that the user can update the schedule from `CRON_TZ=UTC 10 * * * *` to `CRON_TZ=UTC 11 * * * *`?

```
if !strings.Contains(oldSchedule, "TZ") && strings.Contains(schedule, "TZ") {
     // append err
     // "should not add TZ"
} else {
    # should we forbid it to update the schedule with TZ?
    if strings.Contains(schedule, "TZ") && oldSchedule != schedule {
        // append err
    }
}
```
I don't think it's necessary, but may help to make the user migrating to timezone filed.

Reply:
Yes, it's intentional, we want them slowly migrate off, not break them. It's explicitly mentioned in the description.